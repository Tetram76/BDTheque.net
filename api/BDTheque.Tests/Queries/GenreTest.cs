namespace BDTheque.Tests.Queries;

public class GenreTest
{
    [Fact]
    public async Task GetGenresTest()
    {
        string response = await TestServices.ExecuteRequestAsync(
            b => b.SetQuery(
                """
                query {
                    genreList(last: 10, where: {initiale: {eq: "A"}}, order: {nom: ASC}) {
                      pageInfo {
                        hasNextPage
                        hasPreviousPage
                      }
                      edges {
                        cursor
                        node {
                          id
                          nom
                        }
                      }
                    }
                }
                """
            )
        );

        response.Should().MatchSnapshot();
    }
}
