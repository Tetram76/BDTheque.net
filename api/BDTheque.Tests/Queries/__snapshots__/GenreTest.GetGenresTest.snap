{
  "data": {
    "genreList": {
      "pageInfo": {
        "hasNextPage": false,
        "hasPreviousPage": false
      },
      "edges": [
        {
          "cursor": "MA==",
          "node": {
            "id": "c91824664a974a16a9f5204a527766c6",
            "nom": "Action"
          }
        },
        {
          "cursor": "MQ==",
          "node": {
            "id": "411b7fb5347445eb8bd54e1f67291438",
            "nom": "Animation"
          }
        },
        {
          "cursor": "Mg==",
          "node": {
            "id": "fdac06e314ba48019064b6d87527f7c6",
            "nom": "Anticipation"
          }
        },
        {
          "cursor": "Mw==",
          "node": {
            "id": "9d8f75cbfeac48af86aef31c737d1b3f",
            "nom": "Aventures"
          }
        }
      ]
    }
  }
}
